app/src/main/java/chat/rocket/android/chatroom/service/MessageService.kt

```diff
@@ -58,10 +67,19 @@ class MessageService : JobService() {
         if (temporaryMessages.isNotEmpty()) {
             val client = factory.create(serverUrl).client
             temporaryMessages.forEach { message ->
+
+                // Check if it is the dummy message object created by us
+                val attachments = message.attachments
+                if((message.message=="") && (attachments !== null) && attachments.size==1 && attachments[0].imageUrl!==null){
```

Comment:
Besides the message issue mentioned in a previous comment, shouldn't this work for any file upload, not just an image?  Or does image url also work for any file uploaded from the device?

Reply:
Added placeholder for other file uploads. A known issues : for files other than images (atleast for PDFs), on clicking the dummy message (not synced with the server) doesn't open up the file in preview mode. Once it syncs with the server, the file is accessible.